packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1414,18 +1419,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // case this invocation doesn't own
         // flushing.
         if (this.flushMode === FlushMode.Manual) {
-            callback();
-        } else {
-            const savedFlushMode = this.flushMode;
+            this.trackOrderedCalls(callback);
+            return;
+        }
 
-            this.setFlushMode(FlushMode.Manual);
+        const savedFlushMode = this.flushMode;
+        this.setFlushMode(FlushMode.Manual);
 
-            try {
-                callback();
-            } finally {
-                this.flush();
-                this.setFlushMode(savedFlushMode);
-            }
+        try {
+            this.trackOrderedCalls(callback);
+        } finally {
+            this.flush();
+            this.setFlushMode(savedFlushMode);
+        }
+    }
+
+    private trackOrderedCalls(callback: () => void): void {
```

Comment:
Consider renaming this to `trackOrderSequentiallyCalls` since this specifically only called for that function.